mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -228,17 +246,50 @@ impl mayastor_server::Mayastor for MayastorSvc {
                     return Err(Status::invalid_argument("Missing devices"));
                 }
 
-                let rx = rpc_submit::<_, _, LvsError>(async move {
-                    let pool = Lvs::create_or_import(args).await?;
-                    // Capture current pool config and export to file.
-                    PoolConfig::capture().export().await;
-                    Ok(Pool::from(pool))
-                })?;
-
-                rx.await
-                    .map_err(|_| Status::cancelled("cancelled"))?
-                    .map_err(Status::from)
-                    .map(Response::new)
+                let resp = match PoolBackend::try_from(args.pooltype)? {
+                    PoolBackend::Lvs => {
+                        if  MayastorFeatures::get_features().lvm {
+                            // check if a lvm pool already exists with the same name
+                            if let Some(_pool) = lookup_lvm_vg(args.name.as_str()).await {
+                                return Err(Status::invalid_argument("lvm pool with the same name already exists"))
+                            };
+                            // check if the disks are used by existing lvm pool
+                            if let Some(_disk) = lookup_lvm_disk(args.disks[0].as_str()).await {
+                                return Err(Status::invalid_argument("a lvm pool already uses the disk"))
+                            };
+                        }
+                        let rx = rpc_submit::<_, _, LvsError>(async move {
+                            let pool = Lvs::create_or_import(args).await?;
+                            // Capture current pool config and export to file.
+                            PoolConfig::capture().export().await;
+                            Ok(Pool::from(pool))
+                        })?;
+                        rx.await
+                            .map_err(|_| Status::cancelled("cancelled"))?
+                            .map_err(Status::from)
+                            .map(Response::new)
+                    },
+                    PoolBackend::Lvm => {
+                        // check if a lvs pool already exists with the same name
+                        if let Some(_pool) = Lvs::lookup(args.name.as_str()) {
+                            return Err(Status::invalid_argument("lvs pool with the same name already exists"))
+                        };
+                        // check if the disks are used by existing lvs pool
+                        if Lvs::iter()
+                            .map(|l| l.base_bdev().name()).any(|d: String| args.disks.contains(&d)){
```

Comment:
is the `String` type annotation needed here? Compiler can't infer? If it can, convention is to not specify.

Reply:
Yes, it can. I was learning iters at this point and forgot to remove it.

Reply:
> Yes, it can. 

Cool.

> I was learning iters at this point and forgot to remove it.

NP, we all make these mistakes all the time. This is why reviews are important. :)